packages/lib/src/basic-io.ts

```diff
@@ -82,8 +80,6 @@ export async function inputAndOutputItem<I, O>(command: SmartThingsCommandInterf
 			await formatAndWriteItem(command, item, defaultIOFormat)
 		}
 	} else {
-		// TODO: should the be handled by buildInputProcessor.hasInput?
-		// throw new CLIError('input is required either via file specified with --input option or from stdin')
 		throw new CLIError('input is required either via file specified with --input option or from stdin')
```

Comment:
That's an interesting idea. Just yesterday I was testing that I at least got the same error out of the "presentation:device-config:generate" command that we got before (the command I have to finish testing before I can remove the WIP) and noticed that the "missing arg" error (generated by oclif) refers you to the help but doesn't display it and I was wishing it would. I wonder if we could somehow more generally fix this by adding a custom error handler or something like that.

Reply:
We can definitely do that with the [oclif provided catch](https://oclif.io/docs/error_handling#error-handling-in-the-catch-method) handler, we would just have to decide which circumstances call for `help` output and which should surface as an error.

Reply:
One thing we could do is create a new error type that has a flag which indicates whether displaying help would be useful or not, at least for errors we generate ourselves.